_posts/2018-02-28-Update-to-Java11.md

```diff
@@ -0,0 +1,124 @@
+
+There is some Docker love coming up in Java 10 - improved container detection and resource configuration usage. So Java becomes more aware of its environment, especially running in containers.
+
+As always, better performance, improved garbage collection, experimental Jit compiler, open sourced root certificates, yada yada yada. Expect the major releases to happen much faster, even twice a year. Long term support releases will come every 3 years - but to get that three years support updates you will probably need to pay for it, too. See the blog link at end of article on free, secure, stable.
+
+JDK 11 will continue the modularity work and more aggressively remove modules from core Java, making it lighter and easier to update. Next on list are CORBA and Java EE modules. You can still get them, on top of core Java, when you need them. But they're not bundled with every micro-service, freeing up 22 megabytes and 9 modules from it. It goes leaner and meaner.
+
+Project Panama and Valhalla are still coming up, value types, generic specialization, native function calling from JVM, etc. Amber project started with local variable type inference, but enums will be having much more love in future versions, too. At this early stage, not much concrete to show yet.
+
+But it's probably safe to say - this is not your grandfather's Java anymore.
+
+## Disclaimer and closing thoughts
+
+You might have noticed that I'm an incurable jester. Half of what I speak is not so serious, trick is detecting which half. I do not endorse you to use always the latest available platform for running production systems, unless you know what you are doing. I certainly do not encourage you to run outdated ones, either. But you should be relatively fine with Java 8 for some years more. So please do not install JDK 11 on your production system yet.
+ 
+I do have great passion for technology and believe firmly in always having eye for future, and playing with things before using them. Java 11 is now in play-phase, and this article is about understanding you can already start learning it and playing with it, to be prepared. 
+
+Moving from Java 8 to any of upcoming releases is not going to be minor update - it's going to be a huge, risky and painful update, mostly due to module system. It's also going to bring marvelous new features and capabilities, with potential to keep the Java platform alive for years to come. 
+
+Interesting thing is that other JVM languages such as Kotlin, Scala, Clojure and Groovy will need to be updated too, and so will numerous libraries that are not compatible or not using the new features available. Since this is a lot of work, it's quite likely that many add-ons on top of Java will simply choke on migration to 9 and die out. It will be interesting to see which ones have vibrant community behind and will survive.
```

Comment:
"**The** interesting thing…" 

"…or aren't using"

"choke on the migration to (version) 9…"

Maybe "have **a** vibrant enough community behind **them** to survive"